src/terms/magnetic.jl

```diff
@@ -50,3 +50,30 @@ function ene_ops(term::TermMagnetic, basis::PlaneWaveBasis{T}, ψ, occupation;
 
     (; E, ops)
 end
+
+"""
+Specialised for AbstractGPUArray
+"""
+function ene_ops(term::TermMagnetic, basis::PlaneWaveBasis{T}, ψ::Vector{A}, occupation;
```

Comment:
That's a lot of code duplication. I'm sure there is a better way to do this. Let's not bother for now. It's really not all that important.

Reply:
Also we put GPU workarounds in src/workarounds/gpu_arrays.jl

Reply:
The problem here is that the GPU version works for CPU, but it is much slower using CPU. The CPU version does not work for GPU... I have moved it to the `workarounds` now.